session/src/test/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandlerTest.java

```diff
@@ -72,7 +72,64 @@ void shouldCreateAndSaveAddressSession()
         Map<String, String> requestHeaders =
                 Map.of(
                         "header-name", "headerValue",
-                        "x-forwarded-for", "192.0.2.0");
+                        "X-Forwarded-For", "192.0.2.0");
+        String subject = "subject";
+        String persistentSessionId = "persistent_session_id_value";
+        String clientSessionId = "govuk_signin_journey_id_value";
+        ArgumentCaptor<AuditEventContext> auditEventContextArgumentCaptor =
+                ArgumentCaptor.forClass(AuditEventContext.class);
+        when(eventProbe.counterMetric(anyString())).thenReturn(eventProbe);
+        when(sessionRequest.getClientId()).thenReturn("ipv-core");
+        when(sessionRequest.getState()).thenReturn("some state");
+        when(sessionRequest.getRedirectUri())
+                .thenReturn(URI.create("https://www.example.com/callback"));
+        when(sessionRequest.hasSharedClaims()).thenReturn(Boolean.TRUE);
+        when(sessionRequest.getSharedClaims()).thenReturn(sharedClaims);
+        when(sessionRequest.getSubject()).thenReturn(subject);
+        when(sessionRequest.getPersistentSessionId()).thenReturn(persistentSessionId);
+        when(sessionRequest.getClientSessionId()).thenReturn(clientSessionId);
+        when(apiGatewayProxyRequestEvent.getBody()).thenReturn("some json");
+        when(apiGatewayProxyRequestEvent.getHeaders()).thenReturn(requestHeaders);
```

Comment:
I think we should match this `when` with a `verify` on the getHeaders as well. Generally I tend to match all the when with corresponding `verify` but I don't think it is necessary here.